File: Exercícios sobre POO/Animais 4-5/Animais em python/animal.py
class Animal:
    def __init__(self, especie, nome):
        self.especie = especie
        self.nome = nome
    
    def emitir_som(self):
        pass
    
def som_todos_animais(lista):
    for animal in lista:
        print(animal.emitir_som())

File: Exercícios sobre POO/Animais 4-5/Animais em java/src/Animal.java
import java.util.ArrayList;

public class Animal {
    private String especie;
    private String nome;

    public Animal(String especie, String nome){
        this.especie = especie;
        this.nome = nome;
    }

    public String emitir_som(){
        return "som de animal";
    }

    public static void som_todos_animais(ArrayList<Animal> lista){
        for(Animal animal : lista){
            System.out.println(animal.emitir_som());
        }
    }
}
